cdk/tooling_environment/stacks/grafana/glue_catalogs_dashboard.template.json

```diff
@@ -15,10 +15,333 @@
+      "id": 6,
+      "options": {
+        "cellHeight": "lg",
+        "footer": {
+          "countRows": false,
+          "fields": "",
+          "reducer": [
+            "sum"
+          ],
+          "show": false
+        },
+        "showHeader": true
+      },
+      "pluginVersion": "10.2.2",
+      "targets": [
+        {
+          "database": "<<DATABASE_NAME>>",
+          "datasource": "Amazon-Timestream",
+          "measure": "count",
+          "rawQuery": "SELECT \r\n    CASE \r\n        WHEN tables_diff > 0 THEN '+ ' || CAST(tables_diff AS VARCHAR)\r\n        WHEN tables_diff = 0 THEN '+ 0'\r\n        ELSE CAST(tables_diff AS VARCHAR)\r\n    END AS \"Tables, diff\",\r\n    CASE \r\n        WHEN partitions_diff > 0 THEN '+ ' || CAST(partitions_diff AS VARCHAR)\r\n        WHEN partitions_diff = 0 THEN '+ 0'\r\n        ELSE CAST(partitions_diff AS VARCHAR)\r\n    END AS \"Partitions, diff\",\r\n    CASE \r\n        WHEN indexes_diff > 0 THEN '+ ' || CAST(indexes_diff AS VARCHAR)\r\n        WHEN indexes_diff = 0 THEN '+ 0'\r\n        ELSE CAST(indexes_diff AS VARCHAR)\r\n    END AS \"Indexes, diff\"\r\nFROM (\r\n    -- Aggregate the differences across tables, partitions, and indexes\r\n    SELECT  \r\n        SUM(tables_count - prev_tables_count) AS tables_diff,\r\n        SUM(partitions_count - prev_partitions_count) AS partitions_diff,\r\n        SUM(indexes_count - prev_indexes_count) AS indexes_diff\r\n    FROM (\r\n        -- Calculate current and previous values for each resource\r\n        SELECT \r\n            time,\r\n            resource_name,\r\n            partitions_count,\r\n            LAG(partitions_count) OVER (PARTITION BY resource_name ORDER BY time) AS prev_partitions_count,\r\n            tables_count,\r\n            LAG(tables_count) OVER (PARTITION BY resource_name ORDER BY time) AS prev_tables_count,\r\n            indexes_count,\r\n            LAG(indexes_count) OVER (PARTITION BY resource_name ORDER BY time) AS prev_indexes_count,\r\n            ROW_NUMBER() OVER (PARTITION BY resource_name ORDER BY time DESC) AS rn\r\n        FROM (\r\n            -- Fetch the latest and earliest rows per resource\r\n            SELECT \r\n                resource_name,\r\n                time,\r\n                partitions_count,\r\n                tables_count,\r\n                indexes_count\r\n            FROM (\r\n                SELECT \r\n                    t.*, \r\n                    ROW_NUMBER() OVER (PARTITION BY resource_name ORDER BY time DESC) AS rn_desc,\r\n                    ROW_NUMBER() OVER (PARTITION BY resource_name ORDER BY time ASC) AS rn_asc\r\n                FROM $__database.$__table t\r\n                WHERE $__timeFilter and resource_name in (${resource_name:singlequote}) and monitored_environment in (${monitored_env:singlequote})\r\n            ) sub_inner\r\n            WHERE rn_desc = 1 OR rn_asc = 1\r\n        ) sub_outer\r\n    ) sub_differences\r\n    WHERE rn = 1\r\n)",
```

Comment:
for readability:
[sql.txt](https://github.com/user-attachments/files/18008301/sql.txt)